File: Daily/2023/1351_countNegativeNumbersInASortedMatrix.cs
public class Solution {
    public int CountNegatives(int[][] grid) {
        int m = grid.Length, n = grid[0].Length;
        int jdx = n - 1, cnt = 0;
        for (int idx = 0; idx < m; idx++) {
            while (
                jdx >= 0 &&
                grid[idx][jdx] < 0
            ) {
                cnt += (m - idx);
                jdx -= 1;
            }
        }

        return cnt;
    }
}

File: Categories/Array/1351_countNegativeNumbersInASortedMatrix.py
from typing import *

class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        jdx, cnt = n - 1, 0
        for idx in range(0, m):
            while jdx >= 0 and grid[idx][jdx] < 0:
                cnt += (m - idx)
                jdx -= 1
            
            
        return cnt